test/e2e/prometheusagent_test.go

```diff
@@ -365,3 +374,249 @@ func testPromAgentReconcileDaemonSetResourceDelete(t *testing.T) {
+		},
+	)
+	require.NoError(t, err)
+
+	app, err := testFramework.MakeDeployment("../../test/framework/resources/basic-app-for-daemonset-test.yaml")
+	require.NoError(t, err)
+
+	err = framework.CreateDeployment(ctx, ns, app)
+	require.NoError(t, err)
+
+	pm := framework.MakeBasicPodMonitor(name)
+	pm.Spec.PodMetricsEndpoints = []monitoringv1.PodMetricsEndpoint{monitoringv1.PodMetricsEndpoint{
+		Port: "web",
+		BasicAuth: &monitoringv1.BasicAuth{
+			Username: v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: name,
+				},
+				Key: "user",
+			},
+			Password: v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: name,
+				},
+				Key: "password",
+			},
+		},
+	}}
+	_, err = framework.MonClientV1.PodMonitors(ns).Create(ctx, pm, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	prometheusAgentDS := framework.MakeBasicPrometheusAgentDaemonSet(ns, name)
+	prometheusAgentDS.Spec.ScrapeInterval = "1s"
+
+	_, err = framework.CreatePrometheusAgentAndWaitUntilReady(ctx, ns, prometheusAgentDS)
+	require.NoError(t, err)
+
+	var pollErr error
+	err = wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 10*time.Minute, false, func(_ context.Context) (bool, error) {
+		ctx := context.Background()
+
+		appPods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{
+			LabelSelector: "group=test",
+		})
+		if err != nil {
+			pollErr = fmt.Errorf("can't list app pods: %w", err)
+			return false, nil
+		}
+
+		appPodsNodes := make([]string, 0, 2)
+		appPodsIPs := make([]string, 0, 2)
+		for _, pod := range appPods.Items {
+			appPodsNodes = append(appPodsNodes, pod.Spec.NodeName)
+			appPodsIPs = append(appPodsIPs, pod.Status.PodIP)
+		}
+
+		paPods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, pa.ListOptions(name))
+		if err != nil {
+			pollErr = fmt.Errorf("can't list prometheus agent pods: %w", err)
+			return false, nil
+		}
+
+		paPodsNodes := make([]string, 0, 2)
+		for _, pod := range paPods.Items {
+			paPodsNodes = append(paPodsNodes, pod.Spec.NodeName)
+		}
+
+		if len(appPodsNodes) != len(paPodsNodes) {
+			pollErr = fmt.Errorf("got %d application pods and %d prometheus-agent pods", len(appPodsNodes), len(paPodsNodes))
+			return false, nil
+		}
+		for _, n := range appPodsNodes {
+			if !slices.Contains(paPodsNodes, n) {
+				pollErr = fmt.Errorf("no prometheus-agent pod found on node %s", n)
+				return false, nil
+			}
+		}
+
+		cfg := framework.RestConfig
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+
+		closer, err := testFramework.StartPortForward(ctx, cfg, "https", paPods.Items[0].Name, ns, "9090")
+		if err != nil {
+			pollErr = fmt.Errorf("can't start port forward to first prometheus agent pod: %w", err)
+			return false, nil
+		}
+		defer closer()
+
+		req, err := http.NewRequestWithContext(ctx, "GET", "https://localhost:9090/api/v1/targets", nil)
```

Comment:
we should use http instead of https.
```suggestion
		req, err := http.NewRequestWithContext(ctx, "GET", "http://localhost:9090/api/v1/targets", nil)
```